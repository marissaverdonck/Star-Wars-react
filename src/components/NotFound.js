import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Header from './Header';

const AppStyle = styled.main`
  background-color: ${({ theme }) => theme.secondaryColor};
  min-width: 100vw;
  min-height: 100vh;
  margin: 0;
  padding: 0;
`;

const Title = styled.h1`
  font-size: ${({ theme }) => theme.fontSizeS};
  font-family: ${({ theme }) => theme.fontTitles};
  color: ${({ theme }) => theme.primaryColorLight};
  padding: ${({ theme }) => theme.indentS};
  word-break: break-word;
  margin: 0 auto;
  max-width: ${({ theme }) => theme.sectionWidth};
`;
const GoBack = styled.div`
  margin: 0 auto;
  max-width: ${({ theme }) => theme.sectionWidth};
  a {
    font-size: ${({ theme }) => theme.fontSizeS};
    font-family: ${({ theme }) => theme.fontTitles};
    color: ${({ theme }) => theme.primaryColor};
    padding: ${({ theme }) => theme.indentS};
    word-break: break-word;
    text-decoration: none;
    :hover {
      text-decoration: underline;
    }
  }
`;

function NotFound({ title, size, message, link }) {
  return (
    <AppStyle>
      <Header size={size} title={title} />
      <Title>{message}</Title>
      <GoBack>{link}</GoBack>
    </AppStyle>
  );
}

NotFound.propTypes = {
  title: PropTypes.string,
  size: PropTypes.string,
  message: PropTypes.string,
  link: PropTypes.element
};

NotFound.defaultProps = {
  title: 'Star Wars Universe.',
  size: 'big',
  message: '404, page not found.',
  link: <Link to="/"> Go back to the homepage &rarr; </Link>
};

export default NotFound;
