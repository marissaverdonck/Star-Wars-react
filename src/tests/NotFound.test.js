import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import NotFound from '../components/NotFound';

it('Renders correctly when there is no title , size and/or message', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a title', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound title="Marissa" />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a size', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound size="big" />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a message', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound message="hello" />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a unknown title', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound title={null} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a array as title', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound title={[1, 2, 3]} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a unknown size', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound size={null} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a array as size', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound size={[1, 2, 3]} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a unknown message', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound message={null} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a array as message', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound message={[1, 2, 3]} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
