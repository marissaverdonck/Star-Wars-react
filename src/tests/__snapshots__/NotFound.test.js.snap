// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly when there is a array as message 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size="big"
      >
        Star Wars Universe.
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  >
    1
    2
    3
  </h1>
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;

exports[`Renders correctly when there is a array as size 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size={
          Array [
            1,
            2,
            3,
          ]
        }
      >
        Star Wars Universe.
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  >
    404, page not found.
  </h1>
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;

exports[`Renders correctly when there is a array as title 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size="big"
      >
        1
        2
        3
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  >
    404, page not found.
  </h1>
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;

exports[`Renders correctly when there is a message 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size="big"
      >
        Star Wars Universe.
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  >
    hello
  </h1>
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;

exports[`Renders correctly when there is a size 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size="big"
      >
        Star Wars Universe.
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  >
    404, page not found.
  </h1>
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;

exports[`Renders correctly when there is a title 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size="big"
      >
        Marissa
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  >
    404, page not found.
  </h1>
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;

exports[`Renders correctly when there is a unknown message 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size="big"
      >
        Star Wars Universe.
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  />
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;

exports[`Renders correctly when there is a unknown size 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size={null}
      >
        Star Wars Universe.
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  >
    404, page not found.
  </h1>
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;

exports[`Renders correctly when there is a unknown title 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size="big"
      >
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  >
    404, page not found.
  </h1>
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;

exports[`Renders correctly when there is no title , size and/or message 1`] = `
<main
  className="sc-dlnjPT iaiIRM"
>
  <header
    className="sc-bdnylx boUdXo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-gtssRu ktzkmZ"
        size="big"
      >
        Star Wars Universe.
         
      </h1>
    </a>
  </header>
  <h1
    className="sc-hKFyIo ggVltP"
  >
    404, page not found.
  </h1>
  <div
    className="sc-eCApGN cxpRRu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Go back to the homepage → 
    </a>
  </div>
</main>
`;
